### PR TITLE
Added ProtobufSerialization Documentation

### PR DESCRIPTION
**Pull request by nschnitzer**
### Contains

Adds Documentation to ProtobufSerialization

### How to test

Ensure there are no build problems. 

### Outstanding before merging

N/A

### CODE REVIEW

**Comment by jellysnake** [comment]
Note: This PR is a continuation of #3558

File: engine/src/main/java/org/terasology/persistence/serializers/ProtobufSerializer.java [hunk 1]
```diff
@@ -30,42 +30,101 @@
 import java.io.InputStream;
 import java.io.OutputStream;
 
-// TODO: Document
```
Comment (jellysnake) [inline engine/src/main/java/org/terasology/persistence/serializers/ProtobufSerializer.java]: All the javadoc for the methods looks good. I'd just like to see some on the class itself stating what it does. If you need help please ask on the Discord and a mentor can give you some assistance

Comment (jellysnake) [inline engine/src/main/java/org/terasology/persistence/serializers/ProtobufSerializer.java]: All the javadoc for the methods looks good. I'd just like to see some on the class itself stating what it does. If you need help please ask on the Discord and a mentor can give you some assistance

Comment (nschnitzer) [reply engine/src/main/java/org/terasology/persistence/serializers/ProtobufSerializer.java]: I took a look at other serializers in the package and based my class documentation formatting off that.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**